<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>FlightController</class>
 <widget class="QMainWindow" name="FlightController">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>780</width>
    <height>550</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>FlightController</string>
  </property>
  <widget class="QWidget" name="centralWidget">
   <layout class="QHBoxLayout" name="horizontalLayout">
    <item>
     <layout class="QGridLayout" name="gridLayout">
      <item row="27" column="0">
       <widget class="QPushButton" name="CopyAllButton">
        <property name="text">
         <string>CopyAll</string>
        </property>
       </widget>
      </item>
      <item row="6" column="0">
       <widget class="QPushButton" name="delaysButton">
        <property name="text">
         <string>Delays</string>
        </property>
       </widget>
      </item>
      <item row="5" column="0">
       <widget class="QPushButton" name="layerModsButton">
        <property name="text">
         <string>Layer mods</string>
        </property>
       </widget>
      </item>
      <item row="7" column="0">
       <widget class="QPushButton" name="macrosButton">
        <property name="text">
         <string>Macros</string>
        </property>
       </widget>
      </item>
      <item row="13" column="0">
       <widget class="QPushButton" name="thresholdsButton">
        <property name="text">
         <string>Thresholds</string>
        </property>
       </widget>
      </item>
      <item row="12" column="0">
       <widget class="QPushButton" name="KeyMonitorButton">
        <property name="text">
         <string>Key monitor</string>
        </property>
       </widget>
      </item>
      <item row="28" column="0">
       <widget class="QPushButton" name="ClearButton">
        <property name="text">
         <string>Clear</string>
        </property>
       </widget>
      </item>
      <item row="25" column="0">
       <widget class="QPushButton" name="reconnectButton">
        <property name="text">
         <string>Reconnect</string>
        </property>
       </widget>
      </item>
      <item row="4" column="0">
       <widget class="QPushButton" name="layoutButton">
        <property name="text">
         <string>Layout</string>
        </property>
       </widget>
      </item>
      <item row="24" column="0">
       <widget class="QPushButton" name="BootloaderButton">
        <property name="text">
         <string>Update FW</string>
        </property>
       </widget>
      </item>
      <item row="26" column="0">
       <spacer name="verticalSpacer">
        <property name="orientation">
         <enum>Qt::Vertical</enum>
        </property>
        <property name="sizeHint" stdset="0">
         <size>
          <width>20</width>
          <height>40</height>
         </size>
        </property>
       </spacer>
      </item>
      <item row="8" column="0">
       <spacer name="verticalSpacer_2">
        <property name="orientation">
         <enum>Qt::Vertical</enum>
        </property>
        <property name="sizeHint" stdset="0">
         <size>
          <width>20</width>
          <height>40</height>
         </size>
        </property>
       </spacer>
      </item>
      <item row="22" column="0">
       <spacer name="verticalSpacer_3">
        <property name="orientation">
         <enum>Qt::Vertical</enum>
        </property>
        <property name="sizeType">
         <enum>QSizePolicy::MinimumExpanding</enum>
        </property>
        <property name="sizeHint" stdset="0">
         <size>
          <width>20</width>
          <height>13</height>
         </size>
        </property>
       </spacer>
      </item>
      <item row="11" column="0">
       <widget class="QPushButton" name="hwButton">
        <property name="text">
         <string>Hardware</string>
        </property>
       </widget>
      </item>
     </layout>
    </item>
    <item>
     <layout class="QGridLayout" name="gridLayout_5">
      <property name="spacing">
       <number>0</number>
      </property>
      <item row="1" column="9">
       <widget class="QLabel" name="degreesC">
        <property name="text">
         <string>Â°C </string>
        </property>
       </widget>
      </item>
      <item row="1" column="8">
       <widget class="QLabel" name="tempGauge">
        <property name="text">
         <string>+30</string>
        </property>
       </widget>
      </item>
      <item row="1" column="10">
       <widget class="QLabel" name="txLabel">
        <property name="text">
         <string> TX </string>
        </property>
       </widget>
      </item>
      <item row="1" column="0">
       <widget class="QLabel" name="swVersionLegend">
        <property name="text">
         <string>FC v</string>
        </property>
       </widget>
      </item>
      <item row="1" column="13">
       <widget class="QLabel" name="rxSpacer">
        <property name="text">
         <string/>
        </property>
       </widget>
      </item>
      <item row="1" column="12">
       <widget class="QLabel" name="rxLabel">
        <property name="text">
         <string> RX </string>
        </property>
       </widget>
      </item>
      <item row="1" column="14">
       <widget class="QToolButton" name="scanButton">
        <property name="enabled">
         <bool>true</bool>
        </property>
        <property name="sizePolicy">
         <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
          <horstretch>0</horstretch>
          <verstretch>0</verstretch>
         </sizepolicy>
        </property>
        <property name="text">
         <string>Scan</string>
        </property>
        <property name="toolButtonStyle">
         <enum>Qt::ToolButtonTextOnly</enum>
        </property>
        <property name="autoRaise">
         <bool>true</bool>
        </property>
       </widget>
      </item>
      <item row="1" column="18">
       <widget class="QLabel" name="insaneLabel">
        <property name="text">
         <string> UNSAFE </string>
        </property>
       </widget>
      </item>
      <item row="1" column="15">
       <widget class="QToolButton" name="outputButton">
        <property name="text">
         <string>Output</string>
        </property>
        <property name="autoRaise">
         <bool>true</bool>
        </property>
       </widget>
      </item>
      <item row="1" column="16">
       <widget class="QToolButton" name="setupButton">
        <property name="text">
         <string>Setup</string>
        </property>
        <property name="autoRaise">
         <bool>true</bool>
        </property>
       </widget>
      </item>
      <item row="1" column="11">
       <widget class="QLabel" name="txSpacer">
        <property name="text">
         <string/>
        </property>
       </widget>
      </item>
      <item row="1" column="6">
       <spacer name="horizontalSpacer">
        <property name="orientation">
         <enum>Qt::Horizontal</enum>
        </property>
        <property name="sizeHint" stdset="0">
         <size>
          <width>40</width>
          <height>20</height>
         </size>
        </property>
       </spacer>
      </item>
      <item row="1" column="2">
       <widget class="QLabel" name="fwVersionLegend">
        <property name="text">
         <string>, CS v</string>
        </property>
       </widget>
      </item>
      <item row="1" column="3">
       <widget class="QLabel" name="fwVersionLabel">
        <property name="text">
         <string>-.-</string>
        </property>
       </widget>
      </item>
      <item row="1" column="17">
       <widget class="QLabel" name="monitorLabel">
        <property name="text">
         <string>  Monitor  </string>
        </property>
       </widget>
      </item>
      <item row="0" column="0" colspan="19">
       <widget class="LogViewer" name="LogViewport">
        <property name="font">
         <font>
          <family>Courier</family>
          <kerning>false</kerning>
         </font>
        </property>
        <property name="focusPolicy">
         <enum>Qt::WheelFocus</enum>
        </property>
        <property name="acceptDrops">
         <bool>false</bool>
        </property>
        <property name="undoRedoEnabled">
         <bool>false</bool>
        </property>
        <property name="lineWrapMode">
         <enum>QPlainTextEdit::WidgetWidth</enum>
        </property>
        <property name="readOnly">
         <bool>true</bool>
        </property>
        <property name="centerOnScroll">
         <bool>false</bool>
        </property>
       </widget>
      </item>
      <item row="1" column="1">
       <widget class="QLabel" name="swVersionLabel">
        <property name="text">
         <string>-.-</string>
        </property>
       </widget>
      </item>
      <item row="1" column="7">
       <widget class="QLabel" name="latencyLabel">
        <property name="text">
         <string>0 ms </string>
        </property>
       </widget>
      </item>
      <item row="1" column="4">
       <widget class="QLabel" name="label">
        <property name="text">
         <string>, </string>
        </property>
       </widget>
      </item>
      <item row="1" column="5">
       <widget class="QLabel" name="typeLabel">
        <property name="text">
         <string>UNKNOWN</string>
        </property>
       </widget>
      </item>
     </layout>
    </item>
   </layout>
  </widget>
  <widget class="QToolBar" name="mainToolBar">
   <attribute name="toolBarArea">
    <enum>TopToolBarArea</enum>
   </attribute>
   <attribute name="toolBarBreak">
    <bool>false</bool>
   </attribute>
  </widget>
  <widget class="QMenuBar" name="menuBar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>780</width>
     <height>20</height>
    </rect>
   </property>
   <widget class="QMenu" name="menu_File">
    <property name="title">
     <string>&amp;Config</string>
    </property>
    <addaction name="action_Open"/>
    <addaction name="separator"/>
    <addaction name="action_Upload"/>
    <addaction name="action_Download"/>
    <addaction name="separator"/>
    <addaction name="action_Save"/>
    <addaction name="separator"/>
    <addaction name="separator"/>
   </widget>
   <widget class="QMenu" name="menu_Window">
    <property name="title">
     <string>&amp;Window</string>
    </property>
    <addaction name="action_Key_Monitor"/>
    <addaction name="action_Thresholds"/>
    <addaction name="action_Layer_mods"/>
    <addaction name="action_Layout"/>
    <addaction name="action_Macros"/>
    <addaction name="action_Delays"/>
    <addaction name="action_Hardware"/>
   </widget>
   <widget class="QMenu" name="menuCommands">
    <property name="title">
     <string>&amp;Action</string>
    </property>
    <addaction name="action_Setup_mode"/>
    <addaction name="separator"/>
    <addaction name="action_Commit"/>
    <addaction name="action_Rollback"/>
    <addaction name="separator"/>
    <addaction name="action_Update_Firmware"/>
    <addaction name="actionFirmware_File"/>
    <addaction name="separator"/>
   </widget>
   <addaction name="menu_File"/>
   <addaction name="menu_Window"/>
   <addaction name="menuCommands"/>
  </widget>
  <action name="action_Open">
   <property name="text">
    <string>L&amp;oad from file</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+O</string>
   </property>
  </action>
  <action name="action_Save">
   <property name="text">
    <string>&amp;Save to file</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+S</string>
   </property>
  </action>
  <action name="action_Key_Monitor">
   <property name="text">
    <string>&amp;Key monitor</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+K</string>
   </property>
  </action>
  <action name="action_Thresholds">
   <property name="text">
    <string>&amp;Thresholds</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+T</string>
   </property>
  </action>
  <action name="action_Layout">
   <property name="text">
    <string>&amp;Layout</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+L</string>
   </property>
  </action>
  <action name="action_Upload">
   <property name="text">
    <string>&amp;Upload (try changes)</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+U</string>
   </property>
  </action>
  <action name="action_Download">
   <property name="text">
    <string>&amp;Download (revert last changes)</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+D</string>
   </property>
  </action>
  <action name="action_Commit">
   <property name="text">
    <string>&amp;Commit (to EEPROM)</string>
   </property>
  </action>
  <action name="action_Rollback">
   <property name="text">
    <string>&amp;Rollback (to EEPROM config)</string>
   </property>
  </action>
  <action name="action_Update_Firmware">
   <property name="text">
    <string>&amp;Update Firmware</string>
   </property>
   <property name="toolTip">
    <string>Update Firmware</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+F</string>
   </property>
  </action>
  <action name="action_Layer_mods">
   <property name="text">
    <string>L&amp;ayer mods</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+M</string>
   </property>
  </action>
  <action name="action_Setup_mode">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>&amp;Setup mode</string>
   </property>
  </action>
  <action name="actionFirmware_File">
   <property name="text">
    <string>Firmware &amp;File...</string>
   </property>
  </action>
  <action name="action_Hardware">
   <property name="text">
    <string>&amp;Hardware</string>
   </property>
  </action>
  <action name="action_Delays">
   <property name="text">
    <string>&amp;Delays</string>
   </property>
  </action>
  <action name="action_Macros">
   <property name="text">
    <string>&amp;Macros</string>
   </property>
  </action>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <customwidgets>
  <customwidget>
   <class>LogViewer</class>
   <extends>QPlainTextEdit</extends>
   <header>LogViewer.h</header>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections/>
</ui>
